### PR TITLE
[framework] reduce ink sparkle uniform count.

### LINKED ISSUE #133325: [Impeller] ink_sparkle.frag has too many bindings to run on iOS simulators.
### Is there an existing issue for this?

- [x] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Create a flutter project
2. Create any kind of button/InkResponse/InkWell
3. Set `splashFactory` parameter to `InkSparkle.splashFactory`
4. Run the app on an iOS simulator

### Expected results

Expected result is a Material3 animation that is rendered using sparkles.

### Actual results

Only the last frame is triggered (i.e. an overlay on top of the child widget). When pressing the button these logs are triggered continuously until the last frame is reached:

```
[VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Could not create render pipeline: only 14 constant buffers binding are supported in the simulator but 17 were used
[VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime effect pipeline.
```

### Code sample

<details><summary>Code sample</summary>

```dart

Scaffold(
  body: Center(
    child: InkWell(
      onTap: () {},
      splashFactory: InkSparkle.splashFactory,
      child: const Text('InkSparkle'),
    ),
  ),
)

```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

https://github.com/flutter/flutter/assets/4673179/800deb76-db26-44a8-8c0a-214751b5d717

</details>

### Logs

<details><summary>Logs</summary>

```console
[   +9 ms] executing: /usr/bin/arch -arm64e xcrun simctl launch E2758676-D37E-4CB0-8D54-08A086AD4064 com.example.inkTest
--enable-dart-profiling --enable-checked-mode --verify-entry-points
[ +416 ms] com.example.inkTest: 2579
[        ] Waiting for VM Service port to be available...
[ +298 ms] [VERBOSE-2:FlutterDarwinContextMetalImpeller.mm(37)] Using the Impeller rendering backend.
[  +64 ms] VM Service URL on device: http://127.0.0.1:56836/g7-swZrFFIU=/
[   +1 ms] Caching compiled dill
[  +48 ms] Connecting to service protocol: http://127.0.0.1:56836/g7-swZrFFIU=/
[  +69 ms] Launching a Dart Developer Service (DDS) instance at http://127.0.0.1:0, connecting to VM service at
http://127.0.0.1:56836/g7-swZrFFIU=/.
[  +41 ms] DDS is listening at http://127.0.0.1:56839/ANw0F2oisQ4=/.
[  +30 ms] Successfully connected to service protocol: http://127.0.0.1:56836/g7-swZrFFIU=/
[  +13 ms] DevFS: Creating new filesystem on the device (null)
[  +12 ms] DevFS: Created new filesystem on the device
(file:///Users/blinqipa/Library/Developer/CoreSimulator/Devices/E2758676-D37E-4CB0-8D54-08A086AD4064/data/Containers/Data/Application/9A
26C424-A97B-482F-A6F6-00013808C913/tmp/ink_test1bRIwF/ink_test/)
[        ] Updating assets
[  +65 ms] Syncing files to device iPhone 14 Pro...
[   +1 ms] Compiling dart to kernel with 0 updated files
[        ] Processing bundle.
[        ] <- recompile package:ink_test/main.dart 02d05ba7-ea1e-470d-92b5-a7e678b10c20
[        ] <- 02d05ba7-ea1e-470d-92b5-a7e678b10c20
[   +1 ms] Bundle processing done.
[  +73 ms] Updating files.
[        ] DevFS: Sync finished
[        ] Syncing files to device iPhone 14 Pro... (completed in 77ms)
[        ] Synced 0.0MB.
[        ] <- accept
[   +2 ms] Connected to _flutterView/0x12c82fa20.
[   +5 ms] Flutter run key commands.
[   +1 ms] r Hot reload. 🔥🔥🔥
[        ] R Hot restart.
[        ] h List all available interactive commands.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] A Dart VM Service on iPhone 14 Pro is available at: http://127.0.0.1:56839/ANw0F2oisQ4=/
[ +165 ms] The Flutter DevTools debugger and profiler on iPhone 14 Pro is available at:
                    http://127.0.0.1:9102?uri=http://127.0.0.1:56839/ANw0F2oisQ4=/
[+11244 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Could not create render
pipeline: only 14 constant buffers binding are supported in the simulator but 17 were used
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[   +4 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +14 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[   +6 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +14 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[   +1 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[   +1 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[   +1 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[   +2 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +14 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +17 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +14 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +14 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[ +801 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +17 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +14 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[   +1 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +14 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +18 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +14 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +14 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +17 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +17 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +19 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +13 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +18 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +14 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +17 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +15 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.
[  +16 ms] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to get or create runtime
effect pipeline.
[        ] [VERBOSE-2:validation.cc(49)] Break on 'ImpellerValidationBreak' to inspect point of failure: Failed to render entity.

```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.13.1, on macOS 13.2 22D49 darwin-arm64, locale en-GB)
    • Flutter version 3.13.1 on channel stable at /Users/blinqipa/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e1e47221e8 (2 days ago), 2023-08-22 21:43:18 -0700
    • Engine revision b20183e040
    • Dart version 3.1.0
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/blinqipa/Library/Android/sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android Studio Preview.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version unknown)
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.81.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.70.0
   ~/Desktop/banct-mobile     sign-in-flow !5  flutter doctor -v                                                                                               ✔  09:47:01   
[✓] Flutter (Channel stable, 3.13.1, on macOS 13.2 22D49 darwin-arm64, locale en-GB)
    • Flutter version 3.13.1 on channel stable at /Users/blinqipa/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e1e47221e8 (2 days ago), 2023-08-22 21:43:18 -0700
    • Engine revision b20183e040
    • Dart version 3.1.0
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/blinqipa/Library/Android/sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android Studio Preview.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version unknown)
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.81.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.70.0

[✓] Connected device (4 available)
    • sdk gphone64 arm64 (mobile) • emulator-5554                        • android-arm64  • Android 14 (API 34) (emulator)
    • iPhone 14 Pro (mobile)      • E2758676-D37E-4CB0-8D54-08A086AD4064 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)             • macos                                • darwin-arm64   • macOS 13.2 22D49 darwin-arm64
    • Chrome (web)                • chrome                               • web-javascript • Google Chrome 116.0.5845.110

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/133325

Due to the number of uniforms present, the ink_sparkle shader can't run without post processing (on all target platforms) that the impellerc offline compiler doesn't perform. However, we can't just move the uniforms into a uniform buffer object (UBO) because the Skia backend doesn't support it.

Rather than work around this in the compiler, we can reduce the uniform count by 1) packing four floats into a single vec4 2) removing a uniform for what is effectively a constant.

This should have no visible effects, and if any scubas fail it means I did this wrong 😆

### CODE REVIEW

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

**Review by matanlurey — APPROVED** [review]

**Review by bdero — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
This is waiting on the g3 roll to catch up 🤷

File: packages/flutter/lib/src/material/shaders/ink_sparkle.frag [hunk 1]
```diff
@@ -11,22 +11,19 @@ precision highp float;
...
-layout(location = 9) uniform vec2 u_noise_scale;
-layout(location = 10) uniform float u_noise_phase;
-layout(location = 11) uniform vec2 u_circle1;
-layout(location = 12) uniform vec2 u_circle2;
-layout(location = 13) uniform vec2 u_circle3;
-layout(location = 14) uniform vec2 u_rotation1;
-layout(location = 15) uniform vec2 u_rotation2;
-layout(location = 16) uniform vec2 u_rotation3;
+// u_alpha, u_sparkle_alpha, u_blur, u_radius_scale
+layout(location = 1) uniform vec4 u_composite_1;
```
Comment (chinmaygarde) [inline packages/flutter/lib/src/material/shaders/ink_sparkle.frag:15]: Drive by comment: The brittle-ness of specifying locations explicitly and then matching the same in Dart code seems egregious. The compiler auto-generates these and produces metdata. The C++ side uses this metadata for example. Perhaps we could do the same for Dart?

Comment (jonahwilliams) [reply packages/flutter/lib/src/material/shaders/ink_sparkle.frag:15]: If you want to be the next lemming to walk off the cliff of getting Dart code gen to work, be my guest, however I will not be working on this 😄

Comment (chinmaygarde) [reply packages/flutter/lib/src/material/shaders/ink_sparkle.frag:15]: I was thinking of something along the lines of JSON file that is read at runtime that contains a mapping between uniform names and their locations.

But, I must admit I haven't been burned by attempted to get Dart code-gen to work... yet. Perhaps why we can document why it sucks. We are generating code in C++ and I'd say it has worked... fine? Are we talking about specific challenges with Dart that will make it way harder than the engine C++ setup?

**Comment by jonahwilliams** [comment]
The google testing shard has continued to fail due to timeouts on g3 infra. I've marked it as passing and will land it.

**Pull request merged**